web/src/edit/NeighbourhoodMode.svelte

```diff
@@ -88,26 +118,70 @@
   }
 
   function onClickLine(f: Feature, pt: LngLat) {
-    if (action == "filter") {
+    if (action.kind == "filter") {
       $backend!.addModalFilter(pt, $filterType);
       $mutationCounter++;
-    } else if (action == "oneway") {
+    } else if (action.kind == "oneway") {
       $backend!.toggleTravelFlow(f.properties!.road);
       $mutationCounter++;
+    } else if (action.kind == "turn_restriction") {
+      action.from_road_id = f.properties!.road;
+      action.from_road_name = f.properties!.name || "unnamed road";
+      action.possible_targets = $backend!.getTurnRestrictionTargets(
+        f.properties!.road,
+      );
+      // TODO The blue hover state gets stuck
     }
   }
 
-  function deleteFilter(e: CustomEvent<LayerClickInfo>) {
+  function onMapClick(e: CustomEvent<MapMouseEvent>) {
+    if (action.kind != "turn_restriction") {
+      return;
+    }
+
+    // If we click a blank area, reset some state. Not sure why, but clicking
+    // layers doesn't always prevent a click on the map itself.
+    if (
+      $map!.queryRenderedFeatures(e.detail.point, {
+        layers: ["interior-roads", "turn-restriction-targets"],
+      }).length > 0
+    ) {
+      return;
+    }
+
+    action = startTurnRestrictionAction();
+  }
+
+  function createTurnRestriction(e: CustomEvent<LayerClickInfo>) {
+    if (action.kind == "turn_restriction" && action.from_road_id != null) {
+      let to = e.detail.features[0].properties!.road;
+      $backend!.addTurnRestriction(action.from_road_id, to);
+      $mutationCounter++;
+    }
+    action = { kind: "filter" };
+  }
+
+  function deleteModalFilter(e: CustomEvent<LayerClickInfo>) {
     let f = e.detail.features[0];
     $backend!.deleteModalFilter(f.properties!.road);
     $mutationCounter++;
   }
 
+  function deleteTurnRestriction(e: CustomEvent<LayerClickInfo>) {
+    let f = e.detail.features[0];
+    $backend!.deleteTurnRestriction(
+      f.properties!.intersection,
+      f.properties!.from_road,
+      f.properties!.to_road,
+    );
+    $mutationCounter++;
```

Comment:
A bug with the preview lines...

1) Hover on a TR icon. You see the red lines.
2) Click to delete
3) Those lines remain until you move the mouse

I tried some stuff to force recalculating that GJ when mutationCounter changes, but no luck yet